ios/RNGeth.swift

```diff
@@ -82,16 +109,201 @@ class RNGeth: RCTEventEmitter, GethNewHeadHandlerProtocol {
+        do {
+            guard let keyStore = geth_node.getKeyStore() else {
+                throw RuntimeError("KeyStore not ready")
+            }
+            let data = Data(hexString: data as String)
+            let signerAccount = try geth_node.findAccount(rawAddress: signer as String)!
+            let signerAddress = signerAccount.getAddress()!
+            let signature = try keyStore.signHash(signerAddress, hash: data)
+            resolve([signature.hex(prefixed:true)] as NSObject)
+        } catch let NSErr as NSError {
+            NSLog("@", NSErr)
+            reject(nil, nil, NSErr)
+        }
+    }
+
+    /**
+     * Signs data using with passphrase
+     *
+     * @param data String hex encoded input
+     * @param signer String signer address
+     * @param passphrase String the account passphrase
+     * @param promise Promise
+     * @return Return signed transaction
+     */
+    @objc(signHashPassphrase:signer:passphrase:resolver:rejecter:)
+    func signHashPassphrase(data: NSString, signer: NSString, passphrase: NSString, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
+        do {
+            guard let keyStore = geth_node.getKeyStore() else {
+                throw RuntimeError("KeyStore not ready")
+            }
+            let data = Data(hexString: data as String)
+            let signer = try geth_node.findAccount(rawAddress: signer as String)!
+            let signature = try keyStore.signHashPassphrase(signer, passphrase: passphrase as String, hash: data)
+            resolve([signature.hex(prefixed:true)] as NSObject)
+        } catch let NSErr as NSError {
+            NSLog("@", NSErr)
+            reject(nil, nil, NSErr)
+        }
+    }
+
+    /**
+     * Signs a transaction using an unlocked account
+     *
+     * @param txRLP Data RLP encoded transaction
+     * @param from Stirng signer address
+     * @param promise Promise
+     * @return Return signed transaction
+     */
+    @objc(signTransaction:signer:resolver:rejecter:)
+    func signTransaction(txRLPHex: NSString, signer: NSString, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
+        var error: NSError?
+        do {
+            guard let keyStore = geth_node.getKeyStore() else {
+                throw RuntimeError("KeyStore not ready")
+            }
+            let txRLP = Data(hexString: txRLPHex as String)
+            guard let tx = GethNewTransactionFromRLP(txRLP, &error) else {
+                throw error ?? RuntimeError("Unable to create tx from RLP")
+            }
+            let signer = try geth_node.findAccount(rawAddress: signer as String)!
+            let chainID = GethNewBigInt(geth_node.getNodeConfig()?.ethereumNetworkID ?? 0)
+            let signedTx = try keyStore.signTx(signer, tx: tx, chainID: chainID)
+            let encodedTx = try signedTx.encodeRLP()
```

Comment:
our RLP encoding is different from other geth implementations because we have different transaction parameters
does this account for that?
again I am weary of adding more codepaths which implement this though the ones in ContractKit are similarly fragile

Reply:
The `encodeRLP` function here is calling inside of the `celo-blockchain` / go code. So it's not another implementation, it's what I would imagine to be the most trusted one (inside of the actual blockchain client).